parsons/databases/discover_database.py

```diff
@@ -0,0 +1,79 @@
+import os
+from typing import Optional, Union, Type, List
+
+from parsons.databases.database_connector import DatabaseConnector
+from parsons.databases.redshift import Redshift
+from parsons.databases.mysql import MySQL
+from parsons.databases.postgres import Postgres
+from parsons.google.google_bigquery import GoogleBigQuery
+
+
+def discover_database(
```

Comment:
I think this new implementation avoids the unexpected behaviors so I don't think this needs to be a blocker here.